### PR TITLE
Error on bad lines pyarrow

### PR DESCRIPTION
**Pull request by phofl**
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them

cc @lithomas1 Not sure if that was what the TODO was referring to

### CODE REVIEW

**Comment by jbrockmendel** [comment]
> Not sure if that was what the TODO was referring to

I don't know the specific context of that comment, but many of the `TODO(1.4)` comments have to do with the `pyarrow_skip` mark being changed to pyarrow_xfail

File: pandas/tests/io/parser/test_unsupported.py [hunk 1]
```diff
@@ -135,6 +135,9 @@ def test_pyarrow_engine(self):
         1,2,3,4,"""
 
         for default in pa_unsupported:
+            if default == "on_bad_lines":
```
Comment (lithomas1) [inline pandas/tests/io/parser/test_unsupported.py:138]: Is there a reason that on_bad_lines is skipped here and tested separately?

**Comment by lithomas1** [comment]
> cc @lithomas1 Not sure if that was what the TODO was referring to

The TODO was to uncomment that line from the set which I see you've done here.

File: pandas/tests/io/parser/test_unsupported.py (hunk 1 above)
Comment (phofl) [reply pandas/tests/io/parser/test_unsupported.py]: Good point, missed that. The test was failing before but forget to fix. Adjusted it now accordingly

File: pandas/tests/io/parser/test_unsupported.py [hunk 2]
```diff
@@ -140,10 +140,29 @@ def test_pyarrow_engine(self):
...
+            elif default == "on_bad_lines":
+                kwargs[default] = "warn"
             with pytest.raises(ValueError, match=msg):
                 read_csv(StringIO(data), engine="pyarrow", **kwargs)
+
+    @pytest.mark.parametrize(
+        "kwds",
+        [{"on_bad_lines": "warn"}, {"error_bad_lines": True}, {"warn_bad_lines": True}],
+    )
+    def test_pyarrow_bad_lines_fails(self, pyarrow_parser_only, kwds):
```
Comment (lithomas1) [inline pandas/tests/io/parser/test_unsupported.py:157]: Can this test be removed now? It seems to duplicate the ``test_pyarrow_engine`` test above now.

Comment (phofl) [reply pandas/tests/io/parser/test_unsupported.py:157]: good point, thx

**Review by lithomas1 — APPROVED** [review]

**Comment by jreback** [comment]
thanks @phofl and @lithomas1

**Comment by SysuJayce** [comment]
Can we use pandas' `read_csv` with `pyarrow` engine and skip bad lines like "ArrowInvalid: CSV parse error: Expected 2 columns"?

The default engine's `read_csv` is quite slow while many original args are not supported by `pyarrow` engine, which makes it inconvenient to process large data file.

**Pull request merged**